spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
Oh! I didn't see that coming. While writing specs for `parse_formula` I thought it will be little hard to check each and every token, so I thought about converting all tokens to formula and verify the result. That's why I created `to_basic_formula` but I think I've to resort to something else since it won't be a good idea I think to create a class just for `to_basic_formula` to work.